scripts/cat/history.py

```diff
@@ -416,37 +430,86 @@ def add_scar(self, scar_text, condition=None, other_cat=None):
+        object.
+        :param victim: cat object for the victim
+        :param murderer_id: murderer's cat ID
         """
         if not game.clan:
             return
-        if "is_murderer" not in murderer.history.murder:
-            murderer.history.murder["is_murderer"] = []
+        if "is_murderer" not in self.murder:
+            self.murder["is_murderer"] = []
         if "is_victim" not in victim.history.murder:
             victim.history.murder["is_victim"] = []
 
-        murderer.history.murder["is_murderer"].append(
-            {"victim": victim.ID, "revealed": revealed, "moon": game.clan.age}
+        self.murder["is_murderer"].append(
+            {
+                "victim": victim.ID,
+                "moon": game.clan.age,
+                "revealed": {"to_clan": False, "aware_individuals": []},
+            }
         )
+
         victim.history.murder["is_victim"].append(
             {
-                "murderer": murderer.ID,
-                "revealed": revealed,
-                "text": text,
-                "unrevealed_text": unrevealed_text,
+                "murderer": murderer_id,
+                "revealed": {"to_clan": False, "aware_individuals": []},
                 "moon": game.clan.age,
             }
         )
 
+    def reveal_murder(
+        self,
+        victim,
+        murderer_id,
+        clan_reveal: bool = False,
+        aware_individuals: list = None,
+    ):
+        """
+        This adds reveal information to both the murderer and victim's history. This should be called from the murderer's history.
+        :param victim: cat object for the victim
+        :param clan_reveal: set to True if the whole Clan now knows about the murder
+        :param aware_individuals: if only individual cats are learning about the murder, give a list of their cat objects
+        """
+
+        for murder in self.murder["is_murderer"]:
+            if murder["victim"] == victim.ID:
+                if clan_reveal:
+                    murder["revealed"]["to_clan"] = True
+                else:
+                    murder["revealed"]["aware_individuals"].extend(aware_individuals)
+
+        for murder in victim.history.murder["is_victim"]:
+            if murder["murderer"] == murderer_id:
+                if clan_reveal:
+                    murder["revealed"]["to_clan"] = True
+                else:
+                    murder["revealed"]["aware_individuals"].extend(aware_individuals)
+
+    @staticmethod
+    def get_murder_status_text(murder: dict, Cat) -> str:
+        """
+        Returns the complete murder reveal status text for this cat.
+        :param murder: the murder history to pull status text from
+        :param Cat: cat object
+        """
+        text = ""
+        if murder["revealed"]["to_clan"]:
+            text = i18n.t("cat.history.murder_revealed_to_clan", count=1)
+        else:
```

Comment:
ok it'll be way easier for you to make this change than for me. get rid of the `else` and move everything back out one indent
```suggestion
        if murder["revealed"]["to_clan"]:
            return i18n.t("cat.history.murder_revealed_to_clan", count=1)
```

Reply:
Why? The intention is that if a clan is aware, we don't show the aware individuals list.  I feel like the aware individuals is only important when they're the *only* ones who know.  Or am I missing something here with what you're trying to fix lmao

Reply:
because this is what the code does already 🤣 if this check passes, it runs this i18n block, then the next thing it does after is `return text`. so this just shortcuts that? I think?

less indentation means you have to store less information in your brain, easier to balance cognitively.

Reply:
ahhh okay, yeah this was down to me just not properly reading the suggested code, I didn't realize you wanted a return added there.